scheduler/src/cook/mesos/util.clj

```diff
@@ -108,15 +108,22 @@
 (def ^:const default-job-priority 50)
 
 (defn same-user-task-comparator
-  "Comparator to order same user's tasks"
-  [task1 task2]
-  (letfn [(task->feature-vector [task]
+  "Comparator to order same user's tasks
+
+   If consider-backfilled-jobs? is true, then we treat jobs which were backfilled as being the lowest priority"
+  [consider-backfilled-jobs?]
+  (letfn [(task->feature-vector
+            [task]
             ;; Last two elements are aribitary tie breakers.
             ;; Use :db/id because they guarantee uniqueness for different entities
             ;; (:db/id task) is not sufficient because synthetic task entities don't have :db/id
             ;; This assumes there are at most one synthetic task for a job, otherwise uniqueness invariant will break
-            [(- (:job/priority (:job/_instance task) default-job-priority))
+            [(if (and consider-backfilled-jobs? (:instance/backfilled? task))
```

Comment:
I forgot the conclusion of our discussion last but I feel backfilled should be just the first element of the vector instead of wiping out the original priority information.